test/tests/integration/model-test.js

```diff
@@ -0,0 +1,149 @@
+import {
+  dummyModels,
+  createStore
+} from 'tests/test-helper';
+
+import { Promise } from 'rsvp';
+const { Planet, Moon, Star } = dummyModels;
+
+module('Integration - Model', function(hooks) {
+  let store;
+
+  hooks.beforeEach(function() {
+    const models = { planet: Planet, moon: Moon, star: Star };
+    store = createStore({ models });
+  });
+
+  hooks.afterEach(function() {
+    store = null;
+  });
+
+  test('add new model', function(assert) {
+    Ember.run(() => {
+      const done = assert.async();
+
+      Promise.all([
+        store.addRecord({type: 'star', name: 'The Sun'}),
+        store.addRecord({type: 'moon', name: 'Callisto'}),
+      ])
+      .then(([theSun, callisto]) => {
+        store
+          .addRecord({type: 'planet', name: 'Jupiter', star: theSun, moons: [callisto]})
+          .then(record => {
+            assert.ok(record.get('id'), 'assigned id');
+            assert.equal(record.get('name'), 'Jupiter', 'assigned specified attribute');
+            assert.equal(record.get('atmosphere'), false, 'assigned default value for unspecified attribute');
+            assert.equal(record.get('star'), theSun, 'assigned hasOne');
```

Comment:
The liveQuery `relatedRecords` implementation doesn't currently account for when relationships are included when a record is created.